hail/src/test/scala/is/hail/expr/ir/PruneSuite.scala

```diff
@@ -622,13 +623,14 @@ class PruneSuite extends SparkSuite {
   }
 
   @Test def testMatrixMapRowsRebuild() {
-    val mmr = MatrixMapRows(mr, matrixRefStruct(mr.typ, "sa.c2", "va.r2"),
-      Some(FastIndexedSeq("foo"), FastIndexedSeq()))
+    val mmr = MatrixMapRows(
+      MatrixKeyRowsBy(mr, IndexedSeq.empty),
+      matrixRefStruct(mr.typ, "sa.c2", "va.r2"))
     checkRebuild(mmr, subsetMatrixTable(mmr.typ, "global.g1", "g.e1", "va.foo"),
       (_: BaseIR, r: BaseIR) => {
         val mmr = r.asInstanceOf[MatrixMapRows]
         TypeCheck(mmr.newRow, PruneDeadFields.relationalTypeToEnv(mmr.child.typ), None)
-        mmr.child.asInstanceOf[MatrixRead].typ == subsetMatrixTable(mr.typ, "global.g1", "sa.c2", "va.r2", "g.e1")
+        mmr.child.asInstanceOf[MatrixKeyRowsBy].child.asInstanceOf[MatrixRead].typ == subsetMatrixTable(mr.typ, "global.g1", "sa.c2", "va.r2", "g.e1")
```

Comment:
I assume we need to compare the child of the child instead of just the child because we've added an extra IR layer with the KeyRowsBy operation...

Reply:
Right. The extra `KeyRowsBy` layer is just to ensure the `MatrixMapRows` doesn't modify any key fields. There's probably a nicer fix, by making the new row include the old keys, but I was too confused by how these tests work to see how to do that.